src/lib.rs

```diff
@@ -1 +1,192 @@
+        match Child::new(
+            &self.command,
+            self.inherit_handles.unwrap_or(false),
+            self.current_directory.as_ref(),
+        ) {
+            Ok(child) => Ok(child),
+            Err(err) => Err(err),
+        }
+    }
+
+    pub fn status(&mut self) -> Result<ExitStatus> {
+        match self.spawn() {
+            Ok(child) => Ok(child.wait()),
+            Err(err) => Err(err),
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct Child {
+    process_information: PROCESS_INFORMATION,
+}
+
+impl Child {
+    fn new(
+        command: &OsStr,
+        inherit_handles: bool,
+        current_directory: Option<&PathBuf>,
+    ) -> Result<Self> {
+        unsafe {
+            let mut si = STARTUPINFOW::default();
+            let mut pi = PROCESS_INFORMATION::default();
+
+            si.cb = size_of::<STARTUPINFOW>() as u32;
+
+            let process_creation_flags = PROCESS_CREATION_FLAGS(0);
+
+            let res = if let Some(directory) = current_directory {
+                let directory = directory.as_os_str();
+                windows::Win32::System::Threading::CreateProcessW(
+                    PWSTR::default(),
+                    command,
+                    std::ptr::null() as *const SECURITY_ATTRIBUTES,
+                    std::ptr::null() as *const SECURITY_ATTRIBUTES,
+                    inherit_handles,
+                    process_creation_flags,
+                    std::ptr::null() as *const c_void,
+                    directory,
+                    &si,
+                    &mut pi as *mut PROCESS_INFORMATION,
+                )
+            } else {
+                windows::Win32::System::Threading::CreateProcessW(
+                    PWSTR::default(),
+                    command,
+                    std::ptr::null() as *const SECURITY_ATTRIBUTES,
+                    std::ptr::null() as *const SECURITY_ATTRIBUTES,
+                    inherit_handles,
+                    process_creation_flags,
+                    std::ptr::null() as *const c_void,
+                    PWSTR::default(),
+                    &si,
+                    &mut pi as *mut PROCESS_INFORMATION,
+                )
+            };
+
+            if res.as_bool() {
+                Ok(Self {
+                    process_information: pi,
+                })
+            } else {
+                Err(Error::CreationFailed(get_last_error()))
+            }
+        }
+    }
+
+    pub fn wait(&self) -> ExitStatus {
+        let mut exit_code: u32 = 0;
+
+        unsafe {
+            WaitForSingleObject(self.process_information.hProcess, INFINITE);
+            GetExitCodeProcess(
+                self.process_information.hProcess,
+                &mut exit_code as *mut u32,
+            );
+
+            CloseHandle(self.process_information.hProcess);
+            CloseHandle(self.process_information.hThread);
+        }
+
+        ExitStatus(exit_code)
+    }
```

Comment:
This function can give you completely false results! You need to check the response of every function call you do and raise the appropriate error.

To avoid code duplication here, you could call your own public function.